src/options.c

```diff
@@ -347,40 +347,41 @@ void optionsParse(int argc, char** argv)
             warnx("unrecognised option %s", argv[optind++]);
     }
 
-    if (opt.thumb && opt.outputFile)
-        opt.thumbFile = nameThumbnail(opt.outputFile);
-
     /* So that we can safely be called again */
     optind = 1;
 }
 
-char* nameThumbnail(char* name)
+char* nameThumbnail(const char* name)
```

Comment:
This function can be improved in the following:
- const correctness
- C99: reduce the scope of some variables and use `ptrdiff_t`
- replace malloc with calloc

Proposed patch if you agree:
```C
char* nameThumbnail(const char* name)
{
    const char* const thumbSuffix = "-thumb";
    const size_t thumbPrefixLength = 7;
    const size_t newNameLength = strlen(name) + thumbPrefixLength;
    char *const newName = calloc(1, newNameLength);

    if (!newName)
        err(EXIT_FAILURE, "Unable to allocate thumbnail");

    const char* const extension = strrchr(name, '.');

    if (extension) {
        /* We add one so length includes '\0'*/
        const ptrdiff_t nameLength = (extension - name) + 1;
        strlcpy(newName, name, nameLength);
        strlcat(newName, thumbSuffix, newNameLength);
        strlcat(newName, extension, newNameLength);
    } else
        snprintf(newName, newNameLength, "%s%s", name, thumbSuffix);

    return newName;
}
```

Reply:
Thank you! I definitely agree!
The weird thing is I thought I already made the name variable be a constant.